htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxResponse.java

```diff
@@ -196,96 +249,321 @@ public HtmxResponse and(HtmxResponse otherResponse) {
-        if (this.triggers.isEmpty()) {
-            return Collections.emptyMap();
-        }
-        return new HashMap<>(this.triggersAfterSwap);
+    public String getRetarget() {
+        return retarget;
     }
 
-    String getHeaderRetarget() {
-        return headerRetarget;
+    public boolean isRefresh() {
+        return refresh;
     }
 
-    boolean getHeaderRefresh() {
-        return headerRefresh;
+    public Collection<ModelAndView> getTemplates() {
+        return Collections.unmodifiableCollection(templates);
     }
 
-    String getHeaderRedirect() {
-        return headerRedirect;
+    public Map<String, String> getTriggers() {
+        return Collections.unmodifiableMap(this.triggers);
     }
 
-    String getHeaderPushHistory() {
-        return headerPushHistory;
+    public Map<String, String> getTriggersAfterSettle() {
+        return Collections.unmodifiableMap(this.triggersAfterSettle);
     }
 
-    public String getHeaderReswap() {
-        return headerReswap;
+    public Map<String, String> getTriggersAfterSwap() {
+        return Collections.unmodifiableMap(this.triggersAfterSwap);
+    }
+
+    public static final class Builder {
+
+        private Set<ModelAndView> templates = new LinkedHashSet<>();
+        private Map<String, String> triggers = new HashMap<>();
+        private Map<String, String> triggersAfterSettle = new HashMap<>();
+        private Map<String, String> triggersAfterSwap = new HashMap<>();
+        private String pushUrl;
+        private String redirect;
+        private boolean refresh;
+        private HxSwapType reswap;
+        private String retarget;
+
+        /**
+         * Append the rendered template or fragment.
+         *
+         * @param template must not be {@literal null} or empty.
+         * @return the builder
+         */
+        public Builder template(String template) {
+            Assert.hasText(template, "template should not be blank");
+            if (!templates.stream().anyMatch(mav -> template.equals(mav.getViewName()))) {
+                templates.add(new ModelAndView(template));
+            }
+            return this;
+        }
+
+        /**
+         * Append the rendered template or fragment as a resolved {@link View}.
+         *
+         * @param template must not be {@literal null}.
+         * @return the builder
+         */
+        public Builder template(View template) {
+            Assert.notNull(template, "template should not be null");
+            if (!templates.stream().anyMatch(mav -> template.equals(mav.getView()))) {
+                templates.add(new ModelAndView(template));
+            }
+            return this;
+        }
+
+        /**
+         * Append the rendered template or fragment as a {@link ModelAndView}.
+         *
+         * @param template must not be {@literal null}.
+         * @return the builder
+         */
+        public Builder template(ModelAndView template) {
```

Comment:
I recall there being talk about being able to add separate model that are rendered with their separate views. Is that now in HTMX? That would be cool and align with this. I would support adding the `view` call in addition to template.